tests/gnmi/test_gnmi.py

```diff
@@ -56,9 +47,49 @@ def test_gnmi_authorize_failed_with_invalid_cname(duthosts,
     update_list = ["/sonic-db:APPL_DB/localhost/DASH_VNET_TABLE:@/root/%s" % (file_name)]
     msg = ""
     try:
-        gnmi_set(duthost, ptfhost, [], update_list, [])
+        gnmi_set(duthost, ptfhost, [], update_list, [], cert)
     except Exception as e:
         logger.info("Failed to set: " + str(e))
         msg = str(e)
 
+    dump_gnmi_log(duthost)
+
+    return msg
+
+
+def test_gnmi_authorize_failed_with_invalid_cname(duthosts,
+                                                  rand_one_dut_hostname,
+                                                  ptfhost,
+                                                  setup_invalid_client_cert_cname):
+    '''
+    Verify GNMI native write, incremental config for configDB
+    GNMI set request with invalid path
+    '''
+    duthost = duthosts[rand_one_dut_hostname]
+    msg = gnmi_create_vnet(duthost, ptfhost)
+
+    assert "Unauthenticated" in msg
```

Comment:
msg is "Unauthenticated" for both cases.
Is it possible to determine if this is due to an invalid CNAME or a revoked certificate?

Reply:
Fixed by check gnmi log